### PR TITLE
Status bar tests

### PR DESCRIPTION
**Pull request by Draggu**
**Stack**:
- #25
- #24 ⬅
- #27
- #26
- #22

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]
Looks good for my non-ts-js eye

**Review by integraledelebesgue — APPROVED** [review]
Looks fine (as far as I'm able to tell about TS and plugin :v)

File: test-support/scarb.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+import commandExists = require("command-exists");
```
Comment (mkaput) [inline test-support/scarb.ts:2]: why are you not using es6 imports?

File: test-support/scarb.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+export const is_scarb_available = () =>
+  commandExists("scarb")
+    .then(() => true)
+    .catch(() => false);
```
Comment (mkaput) [inline test-support/scarb.ts:7]: 1. camelCase
2. this sounds like something we might want to cache as a global variable

```suggestion
export const scarbAvailable = commandExistsSync("scarb");
```

File: ui-test-resources/empty/Scarb.lock
Comment (mkaput) [inline ui-test-resources/empty/Scarb.lock:1]: can this directory be put inside `ui-tests`? we're proliferating top-level directories too much imo

also I think it should be called `fixtures`

File: test-support/scarb.ts
Comment (mkaput) [inline test-support/scarb.ts:1]: this directory is not using typescript so let's skip it also for this file

File: ui-test/statusBar.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import { EditorView, StatusBar, VSBrowser, WebElement, Workbench } from "vscode-extension-tester";
+import { expect } from "chai";
+import { is_scarb_available } from "../test-support/scarb";
+import * as path from "path";
+
+describe("Status bar test", function () {
```
Comment (mkaput) [inline ui-test/statusBar.ts:6]:
```suggestion
describe("status bar", function () {
```

the whole idea is that all `describe` + `it` calls in the stack form a proper English sentence describing what behaviour is tested

File: ui-test/statusBar.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    is_scarb = await is_scarb_available();
+
+    await VSBrowser.instance.openResources(path.join("ui-test-resources", "empty"));
+  });
+
+  afterEach(async function () {
+    await new EditorView().closeAllEditors();
+  });
+
+  it("checks if status bar is displaying correct message", async function () {
```
Comment (mkaput) [inline ui-test/statusBar.ts:21]:
```suggestion
  it("displays Cairo toolchain version", async function () {
```

File: ui-test/statusBar.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  });
+
+  it("checks if status bar is disabled", async function () {
+    const settings = await new Workbench().openSettings();
+
+    const setting = await settings.findSetting("Show In Status Bar", "Cairo1");
+    setting.setValue(false);
+
+    // Wait so extension have time to refresh.
+    await new Promise((done) => setTimeout(done, 2000));
```
Comment (mkaput) [inline ui-test/statusBar.ts:51]: 2 seconds is a dramatic test execution time increase. we should wait for the statusbar to disappear with such deadline, not always wait so much

**Review by mkaput — CHANGES_REQUESTED** [review]

**Pull request merged**